game/scripts/vscripts/abilities/sohei/sohei_palm_of_life.lua

```diff
@@ -0,0 +1,233 @@
+sohei_palm_of_life = class( AbilityBaseClass )
+
+LinkLuaModifier( "modifier_sohei_palm_of_life_movement", "abilities/sohei/sohei_palm_of_life.lua", LUA_MODIFIER_MOTION_HORIZONTAL )
+
+--------------------------------------------------------------------------------
+
+function sohei_palm_of_life:CastFilterResultTarget( target )
+	local caster = self:GetCaster()
+
+	if caster == target then
+		return UF_FAIL_CUSTOM
+	end
+
+	local ufResult = UnitFilter(
+		target,
+		DOTA_UNIT_TARGET_HERO,
+		DOTA_UNIT_TARGET_TEAM_FRIENDLY,
+		DOTA_UNIT_TARGET_FLAG_NONE,
+		caster:GetTeamNumber()
+	)
+
+	return ufResult
+end
+
+--------------------------------------------------------------------------------
+
+function sohei_palm_of_life:GetCustomCastErrorTarget( target )
+	if self:GetCaster() == target then
+		return "#dota_hud_error_cant_cast_on_self"
+	end
+
+	return ""
+end
+
+--------------------------------------------------------------------------------
+
+function sohei_palm_of_life:OnHeroCalculateStatBonus()
+	local caster = self:GetCaster()
+
+	if caster:HasScepter() then
+		self:SetHidden( false )
+		if self:GetLevel() <= 0 then
+			self:SetLevel( 1 )
+		end
+	else
+		self:SetHidden( true )
```

Comment:
Shouldn't you also unlevel it? Just in case, also makes it copypastable to a generic baseclass

Reply:
This seems pretty unnecessary, as vanilla Aghs added abilities actually start at level 1 no matter what. This is really more of a quick-and-dirty way of doing it as opposed to just starting the ability already leveled as soon as the hero spawns in.